helix-core/src/main/java/org/apache/helix/controller/rebalancer/topology/Topology.java

```diff
@@ -81,62 +75,43 @@ public Topology(final List<String> allNodes, final List<String> liveNodes,
       throw new HelixException(String.format("Config for instances %s is not found!",
           _allInstances.removeAll(_instanceConfigMap.keySet())));
     }
-
     _clusterConfig = clusterConfig;
-    _types = new LinkedHashSet<>();
```

Comment:
I would prefer to keep it. A LinkedHashSet or ArrayList would be much easier to handle compared with array.

Reply:
Updated.